stan/math/rev/core/vari.hpp

```diff
@@ -180,29 +180,41 @@ using vari = vari_value<double>;
 template <typename T, typename = void>
 class vari_view;
 
+/**
+ * This struct is follows the CRTP for methods common to `vari_view<>` and
+ * `vari_value<Matrix>`.
+ */
 template <typename T>
-class vari_view<T, require_not_plain_type_t<T>> final : public vari_base {
- public:
-  using PlainObject = plain_type_t<T>;
-  using value_type = std::decay_t<T>;  // The underlying type for this class
+struct vari_view_crtp {
```

Comment:
I would prefere some other name for this class. Perhaps `vari_dense_base`? Also you can make it inherit from `vari_base` so the classes derived from this one don't need to.

Reply:
I'd like to keep this seperate since it doesn't actually own or make anything. And mixing this with the virtual methods feels weird. 

Yeah I had a hard time naming this. The issue with `vari_dense_base` is that sparse matrices can also use this. I picked this name because it is for creating the views and is just a crtp pattern for calling the right things.

https://eigen.tuxfamily.org/dox/group__TutorialSparse.html#title7

Reply:
by "feels weird" I just had a hard time wrapping my head around the mix of crtp and virtual stuff

Reply:
Fine, than leave the inheritance as it is. How about `vari_base_eigen` than?

Reply:
Sure that works for me